zipline/assets/_assets.pyx

```diff
@@ -187,37 +201,58 @@ cdef class Asset:
         """
         return cls(**dict_)
 
-    def _is_alive(self, dt, bool normalized):
+    def _is_alive_for_session(self, session_label):
         """
         Returns whether the asset is alive at the given dt.
 
         Parameters
         ----------
-        dt: pd.Timestamp
-            The desired timestamp.
-
-        normalized: boolean
-            Whether the date has already been normalized.  If not, we need
-            to first normalize the date before doing the alive check.  If the
-            date is already normalized, this method runs up to 80% faster.
+        session_label: pd.Timestamp
+            The desired session label to check. (midnight UTC)
 
         Returns
         -------
         boolean: whether the asset is alive at the given dt.
         """
-        cdef int64_t dt_value
         cdef int64_t ref_start
         cdef int64_t ref_end
 
-        if not normalized:
-            dt_value = normalize_date(dt).value
-        else:
-            dt_value = dt.value
-
         ref_start = self.start_date.value
```

Comment:
What calendar's sessions do these boundary values correspond to?  Will futures contracts have start/end dates (session labels) that mean they expire at the close of the exchange they trade on?